hyperjumptech/monika
# Fix Loading Remote Config

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Solves ticket #981 
2. remote configs didn't work properly before returning the error
```console
ERROR: ENOENT: no such file or directory, open 'http://0.0.0.0:8000/monika.yml'
    Error: ENOENT: no such file or directory, open 'http://0.0.0.0:8000/monika.yml'
        at Object.openSync (node:fs:585:3)
        at Object.openSync (pkg/prelude/bootstrap.js:753:32)
        at Object.readFileSync (node:fs:453:35)
        at Object.readFileSync (pkg/prelude/bootstrap.js:1039:36)
```

## How did you implement / how did you fix it  
1.  Improperly passed remote filename to validator
2. Fixed by passing correct file content as param to config validator

## How to test  
1. Call monika with `-c https://raw.githubusercontent.com/hyperjumptech/monika/main/monika.example.yml`
2. Should properly load the remote config

## Screenshot
![image](https://user-images.githubusercontent.com/964106/218655703-3f4dbcd1-2af7-40c4-b362-17e607277a69.png)

`bin/run-dev -c https://gist.githubusercontent.com/dennypradipta/8193f47a6c23ecd263a24b68e69e4494/raw/1cf46b3aa5776166ce3b2e6087876cc66cd9b9f2/monika.yml`
![image](https://user-images.githubusercontent.com/964106/218665812-103e05db-8fd0-4c0b-8222-d83f14401fe2.png)
